tools/flaky_test_monitor/level2/process_summary2_results_test.go

```diff
@@ -0,0 +1,159 @@
+	err = json.Unmarshal(expectedTestSummary2JsonBytes, &expectedTestSummary2)
+	require.Nil(t, err)
+
+	//check all details of test summary level 2 between expected and actual
+	require.Equal(t, len(expectedTestSummary2.TestResults), len(actualTestSummary2.TestResults))
+
+	for expectedTestResultSummaryKey := range expectedTestSummary2.TestResults {
+		expectedTestSummary, isFoundExpected := expectedTestSummary2.TestResults[expectedTestResultSummaryKey]
+		actualTestSummary, isFoundActual := actualTestSummary2.TestResults[expectedTestResultSummaryKey]
+
+		require.True(t, isFoundExpected)
+		require.True(t, isFoundActual)
+
+		require.Equal(t, expectedTestSummary.Package, actualTestSummary.Package)
+		require.Equal(t, expectedTestSummary.Test, actualTestSummary.Test)
+
+		require.Equal(t, expectedTestSummary.Runs, actualTestSummary.Runs)
+		require.Equal(t, expectedTestSummary.Passed, actualTestSummary.Passed)
+		require.Equal(t, expectedTestSummary.Failed, actualTestSummary.Failed)
+		require.Equal(t, expectedTestSummary.Skipped, actualTestSummary.Skipped)
+		require.Equal(t, expectedTestSummary.NoResult, actualTestSummary.NoResult)
+
+		require.Equal(t, expectedTestSummary.FailureRate, actualTestSummary.FailureRate)
+
+		// check all durations
+		require.Equal(t, len(expectedTestSummary.Durations), len(actualTestSummary.Durations))
+		for i := range expectedTestSummary.Durations {
+			require.Equal(t, expectedTestSummary.Durations[i], actualTestSummary.Durations[i])
+		}
+		require.Equal(t, expectedTestSummary.AverageDuration, actualTestSummary.AverageDuration)
+	}
+
+	// make sure calculated summary level 2 is what we expected
+	require.Equal(t, expectedTestSummary2, actualTestSummary2)
+
+	// check failure messages created
+	// there are 2 types of scenarios to test for
+	// 1. test summaries with no failures - these will have a `failures` sub-directory with a
+	//    `.keep` placeholder file (since empty folders can't be added to git)
+	// 2. test summaries with failures - these will have a `failures` sub-directory with failure messages saved
+	//    in text files (1 file/failure under separate sub-directory for each test that has failures)
+
+	// count expected failure directories (1 directory/test)
+	expectedFailureDirEntries, err := os.ReadDir(expectedFailureMessagesPath)
+	require.Nil(t, err)
+
+	// count actual failure directories
+	actualFailureDirEntries, err := os.ReadDir(actualFailureTestDataPath)
+	require.Nil(t, err)
+
+	// expected test summary `failures` directory only has placeholder file with no expected failures
+	if len(expectedFailureDirEntries) == 1 && strings.HasPrefix(expectedFailureDirEntries[0].Name(), ".") {
+		// all expected `failures` sub-directories with no expected failues will have a ".keep" file to force saving empty
+		// directory to git when there are no expected failures - we need to not count this placeholder file
+		require.Equal(t, 0, len(actualFailureDirEntries))
+	} else {
```

Comment:
Go prefers to omit the else, and just have `return` in the preceding if, for these end of function if statements. e.g.:

```
if len(expectedFailureDirEntries) == 1 && strings.HasPrefix(expectedFailureDirEntries[0].Name(), ".") {
		// all expected `failures` sub-directories with no expected failues will have a ".keep" file to force saving empty
		// directory to git when there are no expected failures - we need to not count this placeholder file
		require.Equal(t, 0, len(actualFailureDirEntries))
                return
	}
// Everything in the the else statement currently
```

Reply:
Updated
https://github.com/onflow/flow-go/pull/1707/commits/fd3107ae882392b4360886fde4f776abb4d3ae5b